### ISSUE TITLE
Loss incorporation and learning rate change

### ISSUE THREAD

**Issue by rvk007** [rvk007/DeepNet#7]
Update for getting learning rate changes and Update for loss functions.
